src/object_types/player.lua

```diff
@@ -122,14 +134,42 @@ return {
         end
 
         this.quit_y_counter = this.quit_y_counter + dt
+        this.interact_y_counter = this.interact_y_counter + dt
 
+	-- quit text fade
         if this.pre_quit then
             this.quit_alpha = math.min(this.quit_alpha + dt, 1)
         end
 
         if not this.pre_quit then
             this.quit_alpha = math.max(0, this.quit_alpha - 2 * dt)
         end
+
+
+	-- fade prompt, check if still colliding with interactable
+	-- if pre_interactable is true we know interactable has some value
+        if this.pre_interactable then
+            this.interact_alpha = math.min(this.interact_alpha + dt, 0.8)
+
+            if not util.aabb(char, this.interactable) then
+		this.pre_interactable = false
+	    end
+        end
+
+        if not this.pre_interactable then
+            this.interact_alpha = math.max(0, this.interact_alpha - 2 * dt)
+        end
+
+
+        -- interact prompt fade
+	-- this feels.. wrong..
+	map.foreach_object(function (other_obj)
+            if other_obj ~= this and util.aabb(char, other_obj) then
```

Comment:
This will collide with non-interactable objects as well, such as triggers or effect objects -- maybe we can add an 'interactable' flag on interactable objects?